projects/02/ALU.hdl

```diff
@@ -65,5 +65,14 @@ CHIP ALU {
 
     // if no then out = !out
     Not16(in=XFY, out=notXFY);
-    Mux16(a=XFY, b=notXFY, sel=no, out=out);
+    Mux16(a=XFY, b=notXFY, sel=no, out[0..7]=out07, out[8..15]=out815, out[15]=firstdigit, out=out);
+
+    // if out=0 then zr = 1 else zr = 0
+    Or8Way(in=out07, out=Or8way1);
+    Or8Way(in=out815, out=Or8Way2);
+    Or(a=Or8way1, b=Or8Way2, out=IsOutNotZero);
+    Not(in=IsOutNotZero, out=zr);
+
+    // if out<0 then ng = 1 else ng = 0
+    And(a=firstdigit, b=true, out=ng);

```

Comment:
What’s the purpose of this `And` gate? I don’t understand why it’s here.

Reply:
Removed this noop 😁